### PR TITLE
Fix: Validate project_role

### PR DESCRIPTION
**Pull request by liranfarage89**
### Issue & Steps to Reproduce / Feature Request
While QAing #140 , @RLRabinowitz notice that on PR-Plan we don't validate the role.
### Solution
validate it.

### CODE REVIEW

File: env0/resource_team_project_assignment.go [hunk 1]
```diff
@@ -32,7 +33,14 @@ func resourceTeamProjectAssignment() *schema.Resource {
 				Description: "the assigned role (Admin, Planner, Viewer, Deployer)",
 				Required:    true,
 				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
-					_ = client.Role(val.(string))
+					role := Role(val.(string))
+					if role == "" ||
```
Comment (liranfarage89) [inline env0/resource_team_project_assignment.go:37]: empty validation is already validated in `required`,  omitting it..

**Review by avnerenv0 — APPROVED** [review]
TBH not loving this implementation,
`Role` should probably be a stronger type than `string` and also it's not the responsibility of `resource_team_project_assignment` to validate `Role` values, that should be in the model or similar.

However - not critical AT ALL so definitely approved 👍

**Comment by liranfarage89** [comment]
@avnerenv0 
I've started something(very quickly), but the tests need to be changed, which is the most of work here.

how would you suggest handle that? opening a PR with "doing on the side"?

https://github.com/env0/terraform-provider-env0/tree/chore-move-role-validation (see the last commit)

**Comment by avnerenv0** [comment]
@liranfarage89 

> opening a PR with "doing on the side"?
Nah, if you wanna do it just go ahead and get it over with, not "on the side".
Just let our dev channel know that you want to complete this part, and keep an eye on the ROI of the task, if it get's messy probably leave it.

BTW we talked about breaking that `model` file into file-per-model, you might as well put `Role` in it's own file.

**Comment by liranfarage89** [comment]
that's ROI negative currently, that's why I thought about "doing on the side"  😕

**Pull request merged**